tests/kernel/interrupt/testcase.yaml

```diff
@@ -1,7 +1,14 @@
 tests:
   arch.interrupt:
-    arch_exclude: nios2 riscv32
+    arch_exclude: nios2 riscv32 arc qemu_x86 qemu_x86_coverage qemu_x86_64
     platform_exclude: hexiwear_kw40z frdm_kw41z frdm_kl25z nucleo_f103rb
         nucleo_f091rc olimexino_stm32 usb_kw24d512 stm32_min_dev_blue
         stm32_min_dev_black v2m_beetle
     tags: interrupt
+
+  # Platforms without relevant (gen isr) dynamic interrupt support:
+  arch.interrupt.nodyn:
```

Comment:
I hate this.
Can we just build the test once, with cases ifdefed appropriately based on capabilities?

Reply:
Alas no, because ARC doesn't even build if you enable CONFIG_DYNAMIC_INTERRUPTS.